### PR TITLE
Alert players when admins modify their antag reputation

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
We get enough people asking if they're shadowbanned from antag despite the code not even existing so hopefully this helps curb the inevitable paranoia from the ability to modify antag reputation

### CODE REVIEW

**Comment by Wilchenx** [comment]
Does it protect from catboys and sdql

**Comment by yorii** [comment]
https://atlantaned.space/statbus/stat.php?stat=admin_verb&year=2018&month=3

Are the shadowbans real?

**Comment by ExcessiveUseOfCobblestone** [comment]
Wait until they say this PR was a diversion tactic so admins can shadowban them while they're logged off.

btw everyone I don't like has -999 rep and all my friends have 999, good luck :^)

Really the only answer is to, like admin remarks, have an item that displays their current antag rep. Nothing will be good until they can see the score themselves.

... or just tell them they're dumb and this doesn't happen.

**Comment by Nabski89** [comment]
Doesn't this only tell you when it's messed with when you're logged in?

Is there even any way for players to see what their rep is?

**Comment by CosmicScientist** [comment]
This was the first thing I was going to ask for.

Thanks Kor!

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Needs a way to easily view rep whenever, otherwise you've changed nothing, just shifting the paranoia to "the admins are setting our antag rep when we get off!!!!!"

**Comment by CosmicScientist** [comment]
It isn't changing nothing - this tells you who changed your points when you're on.

Yes seeing your actual points would be a very good feature but it's not like this prevents that.

**Comment by MrStonedOne** [comment]
@KorPhaeron make a new proc, tochat_or_message, that takes a ckey and a message, and puts the line in their chat if they are on, and adds a player message if they are not on.

File: code/modules/admin/verbs/randomverbs.dm [hunk 1]
```diff
@@ -89,6 +89,7 @@
 
 	log_admin("[key_name(usr)]: Modified [key_name(C)]'s antagonist reputation [log_text]")
 	message_admins("<span class='adminnotice'>[key_name_admin(usr)]: Modified [key_name(C)]'s antagonist reputation ([log_text])</span>")
+	to_chat(C, "<span class='userdanger'>[key_name_admin(usr)]: Modified your antagonist reputation ([log_text])</span>")
```
Comment (Jordie0608) [inline code/modules/admin/verbs/randomverbs.dm]: `key_name_admin` will include a PM link and the admin character's name, do you want that?

Comment (Cyberboss) [reply code/modules/admin/verbs/randomverbs.dm]: Should probably just be usr.key

Comment (ExcessiveUseOfCobblestone) [reply code/modules/admin/verbs/randomverbs.dm]: Not that it's a bad thing but this will reveal stealthmins.

**Comment by KorPhaeron** [comment]
@MrStonedOne good idea

**Comment by Shadowflame909** [comment]
>36 modify antag rep.
It has started!!!

**Comment by msgerbs** [comment]
What is antag rep?

**Comment by ishitbyabullet** [comment]
@msgerbs antag rep increases your chance of being antag

so if you had max rep then you're 99.9% likely to get antag of some sort unless it's an antag you don't roll for.

If you had max negative then yeah good fucking luck getting any antag.

**Comment by msgerbs** [comment]
And why would that be used? That seems like nothing good could come of it honestly.

**Comment by MrStonedOne** [comment]
it automatically gets generated for every round, and becoming antag uses it, the idea is the more rounds you go without antag, the more likely you are to get antag, to solve the issue of some people getting antag 5 rounds in a row *(cough sticky)* while others go months without.

This adds a notification for when ever admins manually modify this value to cut down on abuse

**Comment by CitrusGender** [comment]
image

**Comment by Cyberboss** [comment]
>deducted -1000

**Comment by ExcessiveUseOfCobblestone** [comment]
Reminder I run a sid(iscord)e business where 20$ = 999 Antag Rep

_Don't_ tell your friends.

**Comment by KorPhaeron** [comment]
too lazy sorry

**Pull request closed without merging**